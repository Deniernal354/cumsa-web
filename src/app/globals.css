@import "tailwindcss";

:root {
  --background: #f8fafc;
  --foreground: #0f172a;
  --surface: #ffffff;
  --muted: #64748b;
  --navy: #2a7487;
  --blue: #5aced8;
  --red: #c73221;
  --header-top: #123a66;
}

body {
  background: var(--background);
  color: var(--foreground);
  font-family: Arial, Helvetica, sans-serif;
}

a {
  text-underline-offset: 4px;
}

.link-cta {
  color: var(--blue);
  font-weight: 700;
  text-decoration: underline;
  text-underline-offset: 3px;
}

.link-cta:hover {
  color: var(--red);
}

.btn-cozy {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 0.5rem 1rem;
  border-radius: 0.6rem;
  font-weight: 600;
  font-size: 0.9rem;
  transition:
    transform 150ms ease,
    box-shadow 150ms ease,
    opacity 150ms ease;
}

.btn-cozy:hover {
  transform: translateY(-1px);
  box-shadow: 0 6px 16px rgba(0, 0, 0, 0.08);
}

.btn-primary {
  background: var(--blue);
  color: white;
}

.btn-outline {
  background: var(--surface);
  color: var(--foreground);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

:focus-visible {
  outline: 2px solid color-mix(in oklab, var(--blue) 70%, white);
  outline-offset: 2px;
}

.container-base {
  margin-left: auto;
  margin-right: auto;
  max-width: 80rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

@media (min-width: 640px) {
  .container-base {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

@media (min-width: 1024px) {
  .container-base {
    padding-left: 2rem;
    padding-right: 2rem;
  }
}

.transition-base {
  transition: all 200ms ease-out;
}

.card {
  background: var(--surface);
  border: 1px solid rgba(0, 0, 0, 0.12);
  border-radius: 0.75rem;
}

.title {
  color: var(--red);
}

.tint-blue {
  filter: invert(28%) sepia(88%) saturate(3612%) hue-rotate(211deg) brightness(93%) contrast(98%);
}

.logo-mask-blue {
  background-color: var(--blue);
  -webkit-mask-image: url("/CU-MSA-LOGO-blank-2.png");
  mask-image: url("/CU-MSA-LOGO-blank-2.png");
  -webkit-mask-repeat: no-repeat;
  mask-repeat: no-repeat;
  -webkit-mask-position: center;
  mask-position: center;
  -webkit-mask-size: contain;
  mask-size: contain;
  display: inline-block;
}
