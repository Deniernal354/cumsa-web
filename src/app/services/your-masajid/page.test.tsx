import { test, expect } from "@jest/globals";
import { render } from "@testing-library/react";
import Page from "./page";

test("your masajid page snapshot", async () => {
  const ui = await Page();
  const { asFragment } = render(ui);
  expect(asFragment()).toMatchSnapshot();
});
