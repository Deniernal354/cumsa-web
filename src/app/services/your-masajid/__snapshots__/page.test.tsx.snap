// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`your masajid page snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-base py-12"
  >
    <h1
      class="text-3xl font-bold"
    >
      United Muslim Organizations of Ottawa–Gatineau
    </h1>
    <p
      class="mt-4 text-gray-700"
    >
      Explore local masajid and community centers in the Ottawa–Gatineau region.
    </p>
    <div
      class="mt-8 grid gap-8 md:grid-cols-2"
    >
      <div
        class="space-y-3"
      >
        <div
          class="rounded-lg border border-black/10 bg-white"
        >
          <button
            aria-expanded="true"
            class="w-full flex items-center justify-between px-4 py-3 text-left font-medium hover:bg-black/[0.03] transition-base"
          >
            <span>
              Ottawa Mosque (OMA)
            </span>
            <svg
              aria-hidden="true"
              class="w-4 h-4 transition-transform rotate-180"
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                d="M5.23 7.21a.75.75 0 0 1 1.06.02L10 11.116l3.71-3.885a.75.75 0 1 1 1.08 1.04l-4.24 4.44a.75.75 0 0 1-1.08 0L5.21 8.27a.75.75 0 0 1 .02-1.06z"
              />
            </svg>
          </button>
          <div
            class="px-4 pb-4 text-sm text-gray-700 space-y-1"
          >
            <div
              class="font-semibold text-[var(--red)]"
            >
              Information
            </div>
            <div>
              251 Northwestern Ave, Ottawa, ON
            </div>
            <div>
              (613) 722-8763
            </div>
            <div>
              <a
                class="link-cta"
                href="https://www.omaottawa.org/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Website →
              </a>
            </div>
          </div>
        </div>
        <div
          class="rounded-lg border border-black/10 bg-white"
        >
          <button
            aria-expanded="false"
            class="w-full flex items-center justify-between px-4 py-3 text-left font-medium hover:bg-black/[0.03] transition-base"
          >
            <span>
              Masjid ar-Rahmah (AMA)
            </span>
            <svg
              aria-hidden="true"
              class="w-4 h-4 transition-transform "
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                d="M5.23 7.21a.75.75 0 0 1 1.06.02L10 11.116l3.71-3.885a.75.75 0 1 1 1.08 1.04l-4.24 4.44a.75.75 0 0 1-1.08 0L5.21 8.27a.75.75 0 0 1 .02-1.06z"
              />
            </svg>
          </button>
        </div>
        <div
          class="rounded-lg border border-black/10 bg-white"
        >
          <button
            aria-expanded="false"
            class="w-full flex items-center justify-between px-4 py-3 text-left font-medium hover:bg-black/[0.03] transition-base"
          >
            <span>
              South Nepean Muslim Community (SNMC)
            </span>
            <svg
              aria-hidden="true"
              class="w-4 h-4 transition-transform "
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                d="M5.23 7.21a.75.75 0 0 1 1.06.02L10 11.116l3.71-3.885a.75.75 0 1 1 1.08 1.04l-4.24 4.44a.75.75 0 0 1-1.08 0L5.21 8.27a.75.75 0 0 1 .02-1.06z"
              />
            </svg>
          </button>
        </div>
        <div
          class="rounded-lg border border-black/10 bg-white"
        >
          <button
            aria-expanded="false"
            class="w-full flex items-center justify-between px-4 py-3 text-left font-medium hover:bg-black/[0.03] transition-base"
          >
            <span>
              Kanata Muslim Association (KMA)
            </span>
            <svg
              aria-hidden="true"
              class="w-4 h-4 transition-transform "
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                d="M5.23 7.21a.75.75 0 0 1 1.06.02L10 11.116l3.71-3.885a.75.75 0 1 1 1.08 1.04l-4.24 4.44a.75.75 0 0 1-1.08 0L5.21 8.27a.75.75 0 0 1 .02-1.06z"
              />
            </svg>
          </button>
        </div>
        <div
          class="rounded-lg border border-black/10 bg-white"
        >
          <button
            aria-expanded="false"
            class="w-full flex items-center justify-between px-4 py-3 text-left font-medium hover:bg-black/[0.03] transition-base"
          >
            <span>
              Centre Islamique de l’Outaouais (CIO)
            </span>
            <svg
              aria-hidden="true"
              class="w-4 h-4 transition-transform "
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                d="M5.23 7.21a.75.75 0 0 1 1.06.02L10 11.116l3.71-3.885a.75.75 0 1 1 1.08 1.04l-4.24 4.44a.75.75 0 0 1-1.08 0L5.21 8.27a.75.75 0 0 1 .02-1.06z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div>
        <div
          data-testid="stub"
          mosques="[object Object],[object Object],[object Object],[object Object],[object Object]"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
